drivers/slipdev/slipdev.c

```diff
@@ -171,50 +198,27 @@ static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
     if (buf == NULL) {
         if (len > 0) {
             /* remove data */
-            for (; len > 0; len--) {
-                int byte = tsrb_get_one(&dev->inbuf);
-                if ((byte == (int)SLIPDEV_END) || (byte < 0)) {
-                    /* end early if end of packet or ringbuffer is reached;
-                     * len might be larger than the actual packet */
-                    break;
-                }
-            }
+            crb_consume_chunk(&dev->rb, NULL, len);
         } else {
             /* the user was warned not to use a buffer size > `INT_MAX` ;-) */
-            res = (int)tsrb_avail(&dev->inbuf);
+            crb_get_chunk_size(&dev->rb, (size_t*)&res);
         }
     }
     else {
-        int byte;
-        bool escaped = false;
-        uint8_t *ptr = buf;
-
-        do {
-            int tmp;
-
-            if ((byte = tsrb_get_one(&dev->inbuf)) < 0) {
-                /* something went wrong, return error */
-                return -EIO;
-            }
-            tmp = slipdev_unstuff_readbyte(ptr, byte, &escaped);
-            ptr += tmp;
-            res += tmp;
-            if ((unsigned)res > len) {
-                while (byte != SLIPDEV_END) {
-                    /* clear out unreceived packet */
-                    byte = tsrb_get_one(&dev->inbuf);
-                }
-                return -ENOBUFS;
-            }
-        } while (byte != SLIPDEV_END);
+        crb_consume_chunk(&dev->rb, buf, len);
+        res = len;
     }
     return res;
 }
 
 static void _isr(netdev_t *netdev)
 {
+    slipdev_t *dev = (slipdev_t *)netdev;
+
     DEBUG("slipdev: handling ISR event\n");
-    if (netdev->event_callback != NULL) {
+
+    size_t len;
+    while (crb_get_chunk_size(&dev->rb, &len)) {
         DEBUG("slipdev: event handler set, issuing RX_COMPLETE event\n");
         netdev->event_callback(netdev, NETDEV_EVENT_RX_COMPLETE);
```

Comment:
This loop becomes obsolete if #18361 gets merged

Reply:
I guess it's not obsolete anymore?

Reply:
Indeed, we went with the per-driver fix instead of doing it upwards in the stack